engine/src/main/java/com/agiletec/aps/tags/CategoriesTag.java

```diff
@@ -33,42 +33,45 @@
 
 /**
  * Return the list of the system categories.
+ *
  * @author E.Santoboni
  */
 public class CategoriesTag extends TagSupport {
 
 	private static final Logger _logger = LoggerFactory.getLogger(CategoriesTag.class);
-	
+
 	@Override
 	public int doStartTag() throws JspException {
-		ServletRequest request =  this.pageContext.getRequest();
+		ServletRequest request = this.pageContext.getRequest();
 		RequestContext reqCtx = (RequestContext) request.getAttribute(RequestContext.REQCTX);
-    	ICategoryManager catManager = (ICategoryManager) ApsWebApplicationUtils.getBean(SystemConstants.CATEGORY_MANAGER, this.pageContext);
+		ICategoryManager catManager = (ICategoryManager) ApsWebApplicationUtils.getBean(SystemConstants.CATEGORY_MANAGER, this.pageContext);
 		try {
 			List<SelectItem> categories = new ArrayList<SelectItem>();
-			Category root = (null != this.getRoot() && null != catManager.getCategory(this.getRoot())) ? 
-					(Category) catManager.getNode(this.getRoot()) : (Category) catManager.getRoot();
+			Category root = (null != this.getRoot() && null != catManager.getCategory(this.getRoot()))

```

Comment:
Lots of logic in one statement. Either add comment or break up

Reply:
I'll fix it